esp/bindings/http/platform/httptransport.cpp

```diff
@@ -2541,3 +2633,176 @@ void CHttpResponse::handleExceptions(IXslProcessor *xslp, IMultiException *me, c
     if (handleExceptions(xslp, me, serv, meth, errorXslt) && logHandleExceptions)
         PROGLOG("Exception(s) handled");
 }
+
+bool CHttpResponse::shouldCompress(int& compressType)
+{
+    compressType = 0;
+    if (!m_compressionEnabled)
+        return false;
+    if (m_content_length == 0 || m_content.length() == 0)
+        return false;
+
+#ifdef _USE_ZLIB
+    if (hasHeader(HTTP_HEADER_CONTENT_ENCODING) || hasHeader(HTTP_HEADER_TRANSFER_ENCODING)) //Already encoded/compressed
+        return false;
+
+    static const int DEFAULT_MIN_COMPRESS_LENGTH = 1000;
+    int min_len = DEFAULT_MIN_COMPRESS_LENGTH;
+    IPropertyTree* proccfg = nullptr;
+    IEspServer* espserver = queryEspServer();
+    if (espserver)
+        proccfg = espserver->queryProcConfig();
+    if (proccfg)
+        min_len = proccfg->getPropInt("@minCompressLength", DEFAULT_MIN_COMPRESS_LENGTH);
+    if (m_content.length() < min_len)
+        return false;
+
+    IEspContext * context = queryContext();
+    if (!context)
+        return false;
+    CHttpRequest* request = dynamic_cast<CHttpRequest*>(context->queryRequest());
+    if (!request)
+        return false;
+    StringBuffer acceptEncoding;
+    request->getHeader(HTTP_HEADER_ACCEPT_ENCODING, acceptEncoding);
+    if (acceptEncoding.length() == 0)
+        return false;
+    char* aestr = acceptEncoding.toLowerCase().detach();
+    char* nextptr = nullptr;
+    bool found = false;
+    for (char* oneword = j_strtok_r(aestr, ", ", &nextptr); oneword; oneword = j_strtok_r(nullptr, ", ", &nextptr))
+    {
+        if (streq(oneword, "gzip"))
+        {
+            compressType = static_cast<int>(ZlibCompressionType::GZIP);
+            found = true;
+            break; //Use gzip as much as possible due to some reported browser issues with deflate
+        }
+        else if (streq(oneword, "deflate"))
+        {
+            compressType = static_cast<int>(ZlibCompressionType::DEFLATE);
+            found = true;
+        }
+        else if (streq(oneword, "x-deflate"))
+        {
+            compressType = static_cast<int>(ZlibCompressionType::ZLIB_DEFLATE);
+            found = true;
+        }
+    }
+    free(aestr);
+    if (!found)
+        return false;
+
+    const char* content_type = m_content_type.get();
+    if (m_content_type.length() == 0 ||
+         !( strnicmp(content_type, "text", 4) == 0
+         || strnicmp(content_type, HTTP_TYPE_APPLICATION_XML, strlen(HTTP_TYPE_APPLICATION_XML)) == 0
+         || strnicmp(content_type, HTTP_TYPE_SOAP, strlen(HTTP_TYPE_SOAP)) == 0
+         || strnicmp(content_type, HTTP_TYPE_SVG_XML, strlen(HTTP_TYPE_SVG_XML)) == 0
+         || strnicmp(content_type, HTTP_TYPE_JSON, strlen(HTTP_TYPE_JSON)) == 0
+         || strnicmp(content_type, HTTP_TYPE_JAVASCRIPT, strlen(HTTP_TYPE_JAVASCRIPT)) == 0
+         || strnicmp(content_type, HTTP_TYPE_JAVASCRIPT2, strlen(HTTP_TYPE_JAVASCRIPT2)) == 0))
+        return false;
+
+    return true;
+#else
+    return false;
```

Comment:
Because this simplifies everything, I would consider moving this to the very top.
bool CHttpResponse::shouldCompress(int& compressType)
{
#ifndef _USE_ZLIB
  return false;
#else
....
#endif

Reply:
Change made per suggestion.